interpreter/package_test.go

```diff
@@ -0,0 +1,336 @@
+package interpreter_test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux/interpreter"
+	"github.com/influxdata/flux/parser"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+// Implementation of interpreter.Package
+type packageObject struct {
+	name    string
+	value   values.Object
+	effects []values.Value
+}
+
+func (p *packageObject) Name() string {
+	return p.name
+}
+
+func (p *packageObject) PolyType() semantic.PolyType {
+	return p.value.PolyType()
+}
+
+func (p *packageObject) Value() values.Object {
+	return p.value
+}
+
+func (p *packageObject) SideEffects() []values.Value {
+	return p.effects
+}
+
+// Implementation of interpreter.Importer
+type importer struct {
+	packages map[string]interpreter.Package
+}
+
+func (imp *importer) Import(path string) (semantic.Package, bool) {
+	pkg, ok := imp.packages[path]
+	if !ok {
+		return semantic.Package{}, false
+	}
+	return semantic.Package{
+		Name: pkg.Name(),
+		Type: pkg.PolyType(),
+	}, true
+}
+
+func (imp *importer) ImportPackageObject(path string) (interpreter.Package, bool) {
+	pkg, ok := imp.packages[path]
+	return pkg, ok
+}
+
+func TestInterpreter_EvalPackage(t *testing.T) {
+	testcases := []struct {
+		name        string
+		imports     [](map[string]string)
+		program     string
+		want        values.Object
+		sideEffects []values.Value
+	}{
+		{
+			name: "simple",
+			program: `
+package foo
+a = 1
+b = 2.0
+1 + 1
+`,
+			want: values.NewObjectWithValues(
+				map[string]values.Value{
+					"a": values.NewInt(1),
+					"b": values.NewFloat(2.0),
+				}),
+		},
+		{
+			name: "import",
+			imports: []map[string]string{
+				{
+					"path/to/bar": `
+package bar
+x = 10
+`,
+				},
+			},
+			program: ` 
+package foo
+import baz "path/to/bar"
+a = baz.x
+`,
+			want: values.NewObjectWithValues(
+				map[string]values.Value{
+					"a": values.NewInt(10),
+				}),
+		},
```

Comment:
Why does this test not fail? The import is not defined.

Reply:
Each test case has a list of imports that it defines via the `imports` field. Am I missing something?

Reply:
No, at first I thought this test case didn't have any imports defined. I misread it. It looks good. Maybe just a small nit to tab in the strings to be indented with the test case. Since we are not testing formatting or otherwise here it will be easier to read that way.